asab/log.py

```diff
@@ -497,8 +501,7 @@ def _on_read(self):
 		except Exception as e:
 			print("Error on the syslog socket '{}'".format(self._address), e, file=sys.stderr)
 
-		# Close a socket - there is no reason for reading or socket is actually closed
-		self._reset()
```

Comment:
This is the biggest bummer. The `_reset()` method does not reset anything. It discards the socket. For good.
So at some random point, it tries to read from the socket (which is useless) and if the connection is lost, the socket is discarded and self._write_ready is set to False, so the handler stops wrtiting the logs entirely.